akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -274,8 +286,44 @@ object ByteString {
     def asByteBuffers: scala.collection.immutable.Iterable[ByteBuffer] = bytestrings map { _.asByteBuffer }
 
     def decodeString(charset: String): String = compact.decodeString(charset)
+
+    private[akka] def writeToOutputStream(os: OutputStream): Unit =
+      bytestrings.foreach(_.writeToOutputStream(os))
+
+    protected def writeReplace(): AnyRef = new SerializationProxy(this)
   }
 
+  @SerialVersionUID(1L)
+  private class SerializationProxy(@transient private var orig: ByteString) extends Serializable {
```

Comment:
I guess mostly to avoid compatibility issues by forcing all serialization through a single proxy, and this way we can also avoid using vars in the underlying ByteString classes (*if* we wanted to serialize in a different form than as-constructed)